server/models/contests/participate.js

```diff
@@ -0,0 +1,79 @@
+const { pool } = require('../database')
+
+/**
+ * @param {*} param0
+ * @param {Object} param0.params
+ * @param {String} param0.username
+ * @return {Promise}
+ */
+
+function participate({ params, username }) {
+  return new Promise((resolve, reject) => {
+    const { contest_id: contestId } = params
+    pool.getConnection((error, connection) => {
+      if (error) {
+        return reject(error)
+      }
+      connection.beginTransaction((error) => {
+        if (error) {
+          return connection.rollback(() => {
+            connection.release()
+            return reject(error)
+          })
+        }
+        connection.query(
+          `INSERT INTO contests_participants (contest_id, participant) SELECT ?, ? 
+            WHERE (SELECT COUNT(id) FROM user_groups WHERE username=? AND group_id = ANY (SELECT group_id FROM contests_groups WHERE contest_id=?))`,
```

Comment:
Similarly, use EXISTS() instead of COUNT() here.
And better to use EXISTS instead of INSERT INTO... WHERE
The query would be like-
```
INSERT INTO contests_participants (contest_id, participant) VALUES (?, ?) 
            WHERE EXISTS(SELECT 1 FROM user_groups WHERE username=? AND group_id = ANY (SELECT group_id FROM contests_groups WHERE contest_id=?))
```

Reply:
we are using `insert-select` here...so we should not include `VALUES` here...instead, we can do:
```
INSERT INTO contests_participants (contest_id, participant) SELECT ?, ? 
            WHERE EXISTS(SELECT 1 FROM user_groups WHERE username=? AND group_id = ANY (SELECT group_id FROM contests_groups WHERE contest_id=?))
```